Build failure: openmw-tes3mp

### Nixpkgs version

- Unstable (25.05)

### Steps to reproduce

To reproduce:
1. `nix build 'github:nixos/nixpkgs/nixpkgs-unstable#openmw-tes3mp'`

### Can Hydra reproduce this build failure?

Hydra is currently rebuilding this package.

### Link to Hydra build job

_No response_

### Relevant log output

```console
warning: The interpretation of store paths arguments ending in `.drv` recently changed. If this command is now failing try again with '/nix/store/6cl94w6dq9hw90gjk70ara0irk5ks3m9-openmw-tes3mp-unwrapped-0.8.1.drv^*'
Running phase: qtPreHook
@nix { "action": "setPhase", "phase": "qtPreHook" }
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/86gyqyk0hp74pspwdrwr9hz6b0mgr5zm-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
substituteStream() in derivation openmw-tes3mp-unwrapped-0.8.1: WARNING: '--replace' is deprecated, use --replace-{fail,warn,quiet}.>
applying patch /nix/store/7fwh7a6acrk870wq3nz90l150pawbz91-98a7d90ee258ceef9c70b0b2955d0458ec46f048.patch
patching file components/crashcatcher/crashcatcher.cpp
applying patch /nix/store/8n9ny047dd96m13srq0njn3b6lc8x4rb-gcc-13.patch
patching file apps/openmw/mwinput/controlswitch.hpp
patching file components/misc/utf8stream.hpp
patching file components/openmw-mp/Utils.hpp
applying patch /nix/store/hxfgq0pmd19ah0b6y8is9i3grdcchbbv-tes3mp.patch
patching file apps/openmw-mp/Script/Types.hpp
Hunk #1 succeeded at 101 (offset -4 lines).
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
fixing cmake files...
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=O>
CMake Warning (dev) at CMakeLists.txt:1 (project):
  cmake_minimum_required() should be called prior to this top-level project()
  call.  Please see the cmake-commands(7) manual for usage documentation of
  both commands.
This warning is for project developers.  Use -Wno-dev to suppress it.

-- The C compiler identification is GNU 14.2.1
-- The CXX compiler identification is GNU 14.2.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/f0m6caffiykyvsjim9376a3hx2yj2ghj-gcc-wrapper-14.2.1.20250322/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/f0m6caffiykyvsjim9376a3hx2yj2ghj-gcc-wrapper-14.2.1.20250322/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Deprecation Warning at CMakeLists.txt:2 (cmake_minimum_required):
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.


-- Configuring OpenMW...
-- Found RecastNavigation: /nix/store/fcq9359crhqmbxl4g5jicrzdbjv5big8-recastai-unstable-2023-01-02/lib/libRecast.a
-- /nix/store/3ql0hgpsb9xlxph791dj6kq5dnjdlpzf-source/include
-- /nix/store/bx2924v88clbcfw4z4gaqhz6mvh69237-raknet-unstable-2020-01-19/lib/libRakNetLibStatic.a
-- Found RakNet_LIBRARY_RELEASE: /nix/store/bx2924v88clbcfw4z4gaqhz6mvh69237-raknet-unstable-2020-01-19/lib/libRakNetLibStatic.a
-- Found RakNet_INCLUDES: /nix/store/3ql0hgpsb9xlxph791dj6kq5dnjdlpzf-source/include/raknet
-- Found OpenGL: /nix/store/sq50878z0rdxwdyjw3fvbgf6wkpcnpcp-libglvnd-1.7.0/lib/libOpenGL.so
-- Found LZ4: /nix/store/llbxcznfpdzx5gymb1kqjaxw8hksppvj-lz4-1.10.0-lib/lib/liblz4.so
-- Found FFmpeg_AVCODEC 60.31.102
-- Found FFmpeg_AVFORMAT 60.16.100
-- Found FFmpeg_AVUTIL 58.29.100
-- Found FFmpeg_SWSCALE 7.5.100
-- Found FFmpeg_SWRESAMPLE 4.12.100
-- Found FFmpeg: /nix/store/lm3qgjj7150nyr9mhz5pjibqk3kwpbmp-ffmpeg-6.1.2-lib/lib/libavcodec.so;/nix/store/lm3qgjj7150nyr9mhz5pjibqk>
-- Found Bullet: /nix/store/fns7d0ph36kicjcgfzk21m5rx3xswffk-bullet-3.25/lib/libBulletDynamics.so (Required is at least version "286>
-- Checking if Bullet uses double precision
-- Bullet uses double precision
-- Found LIBUNSHIELD 
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Found osgDB: /nix/store/gl6ip4dkba9qiwxczaqnx3jfkdcyx48b-openscenegraph-3.6.5/lib/libosgDB.so
-- Found osgViewer: /nix/store/gl6ip4dkba9qiwxczaqnx3jfkdcyx48b-openscenegraph-3.6.5/lib/libosgViewer.so
-- Found osgText: /nix/store/gl6ip4dkba9qiwxczaqnx3jfkdcyx48b-openscenegraph-3.6.5/lib/libosgText.so
-- Found osgGA: /nix/store/gl6ip4dkba9qiwxczaqnx3jfkdcyx48b-openscenegraph-3.6.5/lib/libosgGA.so
-- Found osgParticle: /nix/store/gl6ip4dkba9qiwxczaqnx3jfkdcyx48b-openscenegraph-3.6.5/lib/libosgParticle.so
-- Found osgUtil: /nix/store/gl6ip4dkba9qiwxczaqnx3jfkdcyx48b-openscenegraph-3.6.5/lib/libosgUtil.so
-- Found osgFX: /nix/store/gl6ip4dkba9qiwxczaqnx3jfkdcyx48b-openscenegraph-3.6.5/lib/libosgFX.so
-- Found osgShadow: /nix/store/gl6ip4dkba9qiwxczaqnx3jfkdcyx48b-openscenegraph-3.6.5/lib/libosgShadow.so
-- Found osgAnimation: /nix/store/gl6ip4dkba9qiwxczaqnx3jfkdcyx48b-openscenegraph-3.6.5/lib/libosgAnimation.so
-- Found osg: /nix/store/gl6ip4dkba9qiwxczaqnx3jfkdcyx48b-openscenegraph-3.6.5/lib/libosg.so
-- Found OpenThreads: /nix/store/gl6ip4dkba9qiwxczaqnx3jfkdcyx48b-openscenegraph-3.6.5/lib/libOpenThreads.so
-- Found OpenSceneGraph: /nix/store/gl6ip4dkba9qiwxczaqnx3jfkdcyx48b-openscenegraph-3.6.5/lib/libosgDB.so;/nix/store/gl6ip4dkba9qiwx>
CMake Warning (dev) at CMakeLists.txt:440 (find_package):
  Policy CMP0167 is not set: The FindBoost module is removed.  Run "cmake
  --help-policy CMP0167" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

This warning is for project developers.  Use -Wno-dev to suppress it.

-- Found Boost: /nix/store/ijdpzk8xcsq77k4g4fy19wr88d6s8pmg-boost-1.87.0-dev/include (found suitable version "1.87.0", minimum requi>
-- Found MyGUI 3.4.2
-- Found SDL2 2.32.52
-- Found OpenAL: /nix/store/q2qhrzww05g9v48rpjnrmkvr6ndxscj7-openal-soft-1.24.2/lib/libopenal.so
-- Found LuaJit 
-- Found LuaJit_LIBRARIES: /nix/store/rgkl9fy5mlv0saaj0bv6hf9qhn8j4m3v-luajit-2.1.1741730670/lib/libluajit-5.1.so
-- Found LuaJit_INCLUDE_DIRS: /nix/store/rgkl9fy5mlv0saaj0bv6hf9qhn8j4m3v-luajit-2.1.1741730670/include
CMake Warning (dev) at /nix/store/i4kysgbsrb6kf5fz2dh7yy4zm9lv3xf5-qtbase-5.15.16-dev/lib/cmake/Qt5Core/Qt5CoreMacros.cmake:44 (mess>
  qt5_use_modules is not part of the official API, and might be removed in Qt
  6.
Call Stack (most recent call first):
  /nix/store/i4kysgbsrb6kf5fz2dh7yy4zm9lv3xf5-qtbase-5.15.16-dev/lib/cmake/Qt5Core/Qt5CoreMacros.cmake:434 (_qt5_warn_deprecated)
  apps/browser/CMakeLists.txt:94 (qt5_use_modules)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE) 
-- Configuring done (8.0s)
-- Generating done (0.3s)
CMake Warning:
  Manually-specified variables were not used by the project:

    BUILD_TESTING
    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CMAKE_INSTALL_BINDIR
    CMAKE_INSTALL_DOCDIR
    CMAKE_INSTALL_INCLUDEDIR
    CMAKE_INSTALL_INFODIR
    CMAKE_INSTALL_LIBDIR
    CMAKE_INSTALL_LIBEXECDIR
    CMAKE_INSTALL_LOCALEDIR
    CMAKE_INSTALL_MANDIR
    CMAKE_INSTALL_SBINDIR
    CMAKE_POLICY_DEFAULT_CMP0025


-- Build files have been written to: /build/source/build
cmake: enabled parallel building
cmake: enabled parallel installing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: -j16 SHELL=/nix/store/9nw8b61s8lfdn8fkabxhbz0s775gjhbr-bash-5.2p37/bin/bash
[  0%] Generating ui_contentselector.h
[  1%] Generating config/moc_settingsbase.cpp
[  1%] Generating config/moc_gamesettings.cpp
[  1%] Generating config/moc_launchersettings.cpp
[  1%] Generating contentselector/model/moc_contentmodel.cpp
[  1%] Generating contentselector/model/moc_loadordererror.cpp
[  1%] Generating contentselector/model/moc_esmfile.cpp
[  1%] Generating contentselector/model/moc_modelitem.cpp
[  1%] Building CXX object extern/osg-ffmpeg-videoplayer/CMakeFiles/osg-ffmpeg-videoplayer.dir/videostate.cpp.o
[  1%] Building CXX object extern/osg-ffmpeg-videoplayer/CMakeFiles/osg-ffmpeg-videoplayer.dir/videoplayer.cpp.o
[  1%] Generating contentselector/view/moc_combobox.cpp
[  1%] Building CXX object extern/oics/CMakeFiles/local_tinyxml.dir/tinyxml.cpp.o
[  2%] Building CXX object extern/oics/CMakeFiles/local_tinyxml.dir/tinyxmlerror.cpp.o
[  2%] Building CXX object extern/oics/CMakeFiles/local_tinyxml.dir/tinyxmlparser.cpp.o
[  2%] Building CXX object extern/osg-ffmpeg-videoplayer/CMakeFiles/osg-ffmpeg-videoplayer.dir/audiodecoder.cpp.o
[  2%] Generating contentselector/model/moc_naturalsort.cpp
/build/source/components/contentselector/model/loadordererror.hpp:0: Note: No relevant classes found. No output generated.
/build/source/components/config/settingsbase.hpp:0: Note: No relevant classes found. No output generated.
/build/source/components/contentselector/model/naturalsort.hpp:0: Note: No relevant classes found. No output generated.
[  2%] Building CXX object extern/oics/CMakeFiles/local_tinyxml.dir/tinystr.cpp.o
[  2%] Generating contentselector/view/moc_contentselector.cpp
[  2%] Generating process/moc_processinvoker.cpp
/build/source/components/config/launchersettings.hpp:0: Note: No relevant classes found. No output generated.
/build/source/components/config/gamesettings.hpp:0: Note: No relevant classes found. No output generated.
[  2%] Building CXX object components/CMakeFiles/components.dir/bsa/memorystream.cpp.o
[  3%] Building CXX object components/CMakeFiles/components.dir/bsa/compressedbsafile.cpp.o
[  3%] Building CXX object components/CMakeFiles/components.dir/vfs/manager.cpp.o
[  3%] Building CXX object components/CMakeFiles/components.dir/vfs/bsaarchive.cpp.o
[  3%] Building CXX object components/CMakeFiles/components.dir/vfs/filesystemarchive.cpp.o
[  3%] Building CXX object components/CMakeFiles/components.dir/resource/scenemanager.cpp.o
[  3%] Building CXX object components/CMakeFiles/components.dir/settings/parser.cpp.o
[  3%] Building CXX object components/CMakeFiles/components.dir/settings/settings.cpp.o
[  3%] Building CXX object components/CMakeFiles/components.dir/resource/keyframemanager.cpp.o
[  3%] Building CXX object components/CMakeFiles/components.dir/bsa/bsa_file.cpp.o
[  3%] Building CXX object components/CMakeFiles/components.dir/vfs/registerarchives.cpp.o
In file included from /build/source/extern/osg-ffmpeg-videoplayer/audiodecoder.cpp:21:
/build/source/extern/osg-ffmpeg-videoplayer/videostate.hpp:88:19: warning: 'AVPacketList' is deprecated [-Wdeprecated-declarations]
   88 |     AVPacketList *first_pkt, *last_pkt;
      |                   ^~~~~~~~~
In file included from /nix/store/ai243axvzm3y9yq7ap2fl8yidn2a3dj7-ffmpeg-6.1.2-dev/include/libavcodec/avcodec.h:44,
                 from /build/source/extern/osg-ffmpeg-videoplayer/audiodecoder.hpp:17,
                 from /build/source/extern/osg-ffmpeg-videoplayer/audiodecoder.cpp:1:
/nix/store/ai243axvzm3y9yq7ap2fl8yidn2a3dj7-ffmpeg-6.1.2-dev/include/libavcodec/packet.h:543:3: note: declared here
  543 | } AVPacketList;
      |   ^~~~~~~~~~~~
/build/source/extern/osg-ffmpeg-videoplayer/videostate.hpp:88:31: warning: 'AVPacketList' is deprecated [-Wdeprecated-declarations]
   88 |     AVPacketList *first_pkt, *last_pkt;
      |                               ^~~~~~~~
/nix/store/ai243axvzm3y9yq7ap2fl8yidn2a3dj7-ffmpeg-6.1.2-dev/include/libavcodec/packet.h:543:3: note: declared here
  543 | } AVPacketList;
      |   ^~~~~~~~~~~~
/build/source/extern/osg-ffmpeg-videoplayer/audiodecoder.cpp: In member function 'void Video::MovieAudioDecoder::setupFormat()':
/build/source/extern/osg-ffmpeg-videoplayer/audiodecoder.cpp:112:50: warning: 'AVCodecContext::channel_layout' is deprecated [-Wdepr>
  112 |     uint64_t inputChannelLayout = mAudioContext->channel_layout;
      |                                                  ^~~~~~~~~~~~~~
/nix/store/ai243axvzm3y9yq7ap2fl8yidn2a3dj7-ffmpeg-6.1.2-dev/include/libavcodec/avcodec.h:1130:14: note: declared here
 1130 |     uint64_t channel_layout;
      |              ^~~~~~~~~~~~~~
/build/source/extern/osg-ffmpeg-videoplayer/audiodecoder.cpp:112:50: warning: 'AVCodecContext::channel_layout' is deprecated [-Wdepr>
  112 |     uint64_t inputChannelLayout = mAudioContext->channel_layout;
      |                                                  ^~~~~~~~~~~~~~
/nix/store/ai243axvzm3y9yq7ap2fl8yidn2a3dj7-ffmpeg-6.1.2-dev/include/libavcodec/avcodec.h:1130:14: note: declared here
 1130 |     uint64_t channel_layout;
      |              ^~~~~~~~~~~~~~
/build/source/extern/osg-ffmpeg-videoplayer/audiodecoder.cpp:112:50: warning: 'AVCodecContext::channel_layout' is deprecated [-Wdepr>
  112 |     uint64_t inputChannelLayout = mAudioContext->channel_layout;
      |                                                  ^~~~~~~~~~~~~~
/nix/store/ai243axvzm3y9yq7ap2fl8yidn2a3dj7-ffmpeg-6.1.2-dev/include/libavcodec/avcodec.h:1130:14: note: declared here
 1130 |     uint64_t channel_layout;
      |              ^~~~~~~~~~~~~~
/build/source/extern/osg-ffmpeg-videoplayer/audiodecoder.cpp:114:75: warning: 'AVCodecContext::channels' is deprecated [-Wdeprecated>
  114 |         inputChannelLayout = av_get_default_channel_layout(mAudioContext->channels);
      |                                                                           ^~~~~~~~
/nix/store/ai243axvzm3y9yq7ap2fl8yidn2a3dj7-ffmpeg-6.1.2-dev/include/libavcodec/avcodec.h:1072:9: note: declared here
 1072 |     int channels;
      |         ^~~~~~~~
/build/source/extern/osg-ffmpeg-videoplayer/audiodecoder.cpp:114:75: warning: 'AVCodecContext::channels' is deprecated [-Wdeprecated>
  114 |         inputChannelLayout = av_get_default_channel_layout(mAudioContext->channels);
      |                                                                           ^~~~~~~~
/nix/store/ai243axvzm3y9yq7ap2fl8yidn2a3dj7-ffmpeg-6.1.2-dev/include/libavcodec/avcodec.h:1072:9: note: declared here
 1072 |     int channels;
      |         ^~~~~~~~
/build/source/extern/osg-ffmpeg-videoplayer/audiodecoder.cpp:114:75: warning: 'AVCodecContext::channels' is deprecated [-Wdeprecated>
  114 |         inputChannelLayout = av_get_default_channel_layout(mAudioContext->channels);
      |                                                                           ^~~~~~~~
/nix/store/ai243axvzm3y9yq7ap2fl8yidn2a3dj7-ffmpeg-6.1.2-dev/include/libavcodec/avcodec.h:1072:9: note: declared here
 1072 |     int channels;
      |         ^~~~~~~~
/build/source/extern/osg-ffmpeg-videoplayer/audiodecoder.cpp:114:59: warning: 'int64_t av_get_default_channel_layout(int)' is deprec>
  114 |         inputChannelLayout = av_get_default_channel_layout(mAudioContext->channels);
      |                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /nix/store/ai243axvzm3y9yq7ap2fl8yidn2a3dj7-ffmpeg-6.1.2-dev/include/libavcodec/avcodec.h:34:
/nix/store/ai243axvzm3y9yq7ap2fl8yidn2a3dj7-ffmpeg-6.1.2-dev/include/libavutil/channel_layout.h:506:9: note: declared here
  506 | int64_t av_get_default_channel_layout(int nb_channels);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/extern/osg-ffmpeg-videoplayer/audiodecoder.cpp:127:51: warning: 'SwrContext* swr_alloc_set_opts(SwrContext*, int64_t, >
  127 |         mAudioResampler->mSwr = swr_alloc_set_opts(mAudioResampler->mSwr,
      |                                 ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
  128 |                           mOutputChannelLayout,
      |                           ~~~~~~~~~~~~~~~~~~~~~    
  129 |                           mOutputSampleFormat,
      |                           ~~~~~~~~~~~~~~~~~~~~     
  130 |                           mOutputSampleRate,
      |                           ~~~~~~~~~~~~~~~~~~       
  131 |                           inputChannelLayout,
      |                           ~~~~~~~~~~~~~~~~~~~      
  132 |                           inputSampleFormat,
      |                           ~~~~~~~~~~~~~~~~~~       
  133 |                           inputSampleRate,
      |                           ~~~~~~~~~~~~~~~~         
  134 |                           0,                             // logging level offset
      |                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  135 |                           nullptr);                      // log context
      |                           ~~~~~~~~                 
In file included from /build/source/extern/osg-ffmpeg-videoplayer/audiodecoder.cpp:14:
/nix/store/ai243axvzm3y9yq7ap2fl8yidn2a3dj7-ffmpeg-6.1.2-dev/include/libswresample/swresample.h:260:20: note: declared here
  260 | struct SwrContext *swr_alloc_set_opts(struct SwrContext *s,
      |                    ^~~~~~~~~~~~~~~~~~
/build/source/extern/osg-ffmpeg-videoplayer/audiodecoder.cpp: In member function 'int Video::MovieAudioDecoder::synchronize_audio()':
/build/source/extern/osg-ffmpeg-videoplayer/audiodecoder.cpp:161:54: warning: 'int av_get_channel_layout_nb_channels(uint64_t)' is d>
  161 |                     av_get_channel_layout_nb_channels(mOutputChannelLayout);
      |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
/nix/store/ai243axvzm3y9yq7ap2fl8yidn2a3dj7-ffmpeg-6.1.2-dev/include/libavutil/channel_layout.h:498:5: note: declared here
  498 | int av_get_channel_layout_nb_channels(uint64_t channel_layout);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/extern/osg-ffmpeg-videoplayer/audiodecoder.cpp: In member function 'int Video::MovieAudioDecoder::audio_decode_frame(A>
/build/source/extern/osg-ffmpeg-videoplayer/audiodecoder.cpp:207:94: warning: 'int av_get_channel_layout_nb_channels(uint64_t)' is d>
  207 |                     if(av_samples_alloc(&mDataBuf, nullptr, av_get_channel_layout_nb_channels(mOutputChannelLayout),
      |                                                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
/nix/store/ai243axvzm3y9yq7ap2fl8yidn2a3dj7-ffmpeg-6.1.2-dev/include/libavutil/channel_layout.h:498:5: note: declared here
  498 | int av_get_channel_layout_nb_channels(uint64_t channel_layout);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/extern/osg-ffmpeg-videoplayer/audiodecoder.cpp:224:79: warning: 'int av_get_channel_layout_nb_channels(uint64_t)' is d>
  224 |             int result = frame->nb_samples * av_get_channel_layout_nb_channels(mOutputChannelLayout) *
      |                                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
/nix/store/ai243axvzm3y9yq7ap2fl8yidn2a3dj7-ffmpeg-6.1.2-dev/include/libavutil/channel_layout.h:498:5: note: declared here
  498 | int av_get_channel_layout_nb_channels(uint64_t channel_layout);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/extern/osg-ffmpeg-videoplayer/audiodecoder.cpp: In member function 'size_t Video::MovieAudioDecoder::read(char*, size_>
/build/source/extern/osg-ffmpeg-videoplayer/audiodecoder.cpp:301:56: warning: 'int av_get_channel_layout_nb_channels(uint64_t)' is d>
  301 |                     * av_get_channel_layout_nb_channels(mOutputChannelLayout);
      |                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
/nix/store/ai243axvzm3y9yq7ap2fl8yidn2a3dj7-ffmpeg-6.1.2-dev/include/libavutil/channel_layout.h:498:5: note: declared here
  498 | int av_get_channel_layout_nb_channels(uint64_t channel_layout);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[  3%] Building CXX object components/CMakeFiles/components.dir/resource/imagemanager.cpp.o
[  3%] Building CXX object components/CMakeFiles/components.dir/resource/bulletshapemanager.cpp.o
[  3%] Building CXX object components/CMakeFiles/components.dir/resource/bulletshape.cpp.o
[  3%] Linking CXX static library liblocal_tinyxml.a
In file included from /build/source/extern/osg-ffmpeg-videoplayer/videostate.cpp:1:
/build/source/extern/osg-ffmpeg-videoplayer/videostate.hpp:88:19: warning: 'AVPacketList' is deprecated [-Wdeprecated-declarations]
   88 |     AVPacketList *first_pkt, *last_pkt;
      |                   ^~~~~~~~~
In file included from /nix/store/ai243axvzm3y9yq7ap2fl8yidn2a3dj7-ffmpeg-6.1.2-dev/include/libavcodec/avcodec.h:44,
                 from /build/source/extern/osg-ffmpeg-videoplayer/videostate.hpp:25:
/nix/store/ai243axvzm3y9yq7ap2fl8yidn2a3dj7-ffmpeg-6.1.2-dev/include/libavcodec/packet.h:543:3: note: declared here
  543 | } AVPacketList;
      |   ^~~~~~~~~~~~
/build/source/extern/osg-ffmpeg-videoplayer/videostate.hpp:88:31: warning: 'AVPacketList' is deprecated [-Wdeprecated-declarations]
   88 |     AVPacketList *first_pkt, *last_pkt;
      |                               ^~~~~~~~
/nix/store/ai243axvzm3y9yq7ap2fl8yidn2a3dj7-ffmpeg-6.1.2-dev/include/libavcodec/packet.h:543:3: note: declared here
  543 | } AVPacketList;
      |   ^~~~~~~~~~~~
[  4%] Building CXX object components/CMakeFiles/components.dir/resource/niffilemanager.cpp.o
[  4%] Built target local_tinyxml
[  4%] Building CXX object extern/oics/CMakeFiles/oics.dir/ICSChannel.cpp.o
[  4%] Building CXX object components/CMakeFiles/components.dir/resource/multiobjectcache.cpp.o
In file included from /build/source/extern/osg-ffmpeg-videoplayer/videoplayer.cpp:8:
/build/source/extern/osg-ffmpeg-videoplayer/videostate.hpp:88:19: warning: 'AVPacketList' is deprecated [-Wdeprecated-declarations]
   88 |     AVPacketList *first_pkt, *last_pkt;
      |                   ^~~~~~~~~
In file included from /nix/store/ai243axvzm3y9yq7ap2fl8yidn2a3dj7-ffmpeg-6.1.2-dev/include/libavcodec/avcodec.h:44,
                 from /build/source/extern/osg-ffmpeg-videoplayer/audiodecoder.hpp:17,
                 from /build/source/extern/osg-ffmpeg-videoplayer/audiofactory.hpp:4,
                 from /build/source/extern/osg-ffmpeg-videoplayer/videoplayer.cpp:7:
/nix/store/ai243axvzm3y9yq7ap2fl8yidn2a3dj7-ffmpeg-6.1.2-dev/include/libavcodec/packet.h:543:3: note: declared here
  543 | } AVPacketList;
      |   ^~~~~~~~~~~~
/build/source/extern/osg-ffmpeg-videoplayer/videostate.hpp:88:31: warning: 'AVPacketList' is deprecated [-Wdeprecated-declarations]
   88 |     AVPacketList *first_pkt, *last_pkt;
      |                               ^~~~~~~~
/nix/store/ai243axvzm3y9yq7ap2fl8yidn2a3dj7-ffmpeg-6.1.2-dev/include/libavcodec/packet.h:543:3: note: declared here
  543 | } AVPacketList;
      |   ^~~~~~~~~~~~
/build/source/components/vfs/filesystemarchive.cpp: In member function 'virtual void VFS::FileSystemArchive::listResources(std::map<>
/build/source/components/vfs/filesystemarchive.cpp:41:22: error: 'transform' is not a member of 'std'
   41 |                 std::transform(proper.begin() + prefix, proper.end(), std::back_inserter(searchable), normalize_function);
      |                      ^~~~~~~~~
[  4%] Building CXX object extern/oics/CMakeFiles/oics.dir/ICSControl.cpp.o
/build/source/extern/osg-ffmpeg-videoplayer/videostate.cpp: In member function 'void Video::PacketQueue::put(AVPacket*)':
/build/source/extern/osg-ffmpeg-videoplayer/videostate.cpp:117:21: warning: 'AVPacketList' is deprecated [-Wdeprecated-declarations]
  117 |     std::unique_ptr<AVPacketList, AVFree<AVPacketList>> pkt1(static_cast<AVPacketList*>(av_malloc(sizeof(AVPacketList))));
      |                     ^~~~~~~~~~~~
/nix/store/ai243axvzm3y9yq7ap2fl8yidn2a3dj7-ffmpeg-6.1.2-dev/include/libavcodec/packet.h:543:3: note: declared here
  543 | } AVPacketList;
      |   ^~~~~~~~~~~~
/build/source/extern/osg-ffmpeg-videoplayer/videostate.cpp:117:42: warning: 'AVPacketList' is deprecated [-Wdeprecated-declarations]
  117 |     std::unique_ptr<AVPacketList, AVFree<AVPacketList>> pkt1(static_cast<AVPacketList*>(av_malloc(sizeof(AVPacketList))));
      |                                          ^~~~~~~~~~~~
/nix/store/ai243axvzm3y9yq7ap2fl8yidn2a3dj7-ffmpeg-6.1.2-dev/include/libavcodec/packet.h:543:3: note: declared here
  543 | } AVPacketList;
      |   ^~~~~~~~~~~~
/build/source/extern/osg-ffmpeg-videoplayer/videostate.cpp:117:86: warning: 'AVPacketList' is deprecated [-Wdeprecated-declarations]
  117 |     std::unique_ptr<AVPacketList, AVFree<AVPacketList>> pkt1(static_cast<AVPacketList*>(av_malloc(sizeof(AVPacketList))));
      |                                                                                      ^
/nix/store/ai243axvzm3y9yq7ap2fl8yidn2a3dj7-ffmpeg-6.1.2-dev/include/libavcodec/packet.h:543:3: note: declared here
  543 | } AVPacketList;
      |   ^~~~~~~~~~~~
/build/source/extern/osg-ffmpeg-videoplayer/videostate.cpp:117:106: warning: 'AVPacketList' is deprecated [-Wdeprecated-declarations]
  117 |     std::unique_ptr<AVPacketList, AVFree<AVPacketList>> pkt1(static_cast<AVPacketList*>(av_malloc(sizeof(AVPacketList))));
      |                                                                                                          ^~~~~~~~~~~~
/nix/store/ai243axvzm3y9yq7ap2fl8yidn2a3dj7-ffmpeg-6.1.2-dev/include/libavcodec/packet.h:543:3: note: declared here
  543 | } AVPacketList;
      |   ^~~~~~~~~~~~
/build/source/extern/osg-ffmpeg-videoplayer/videostate.cpp:128:19: warning: 'AVPacketList' is deprecated [-Wdeprecated-declarations]
  128 |     AVPacketList* ptr = pkt1.release();
      |                   ^~~
/nix/store/ai243axvzm3y9yq7ap2fl8yidn2a3dj7-ffmpeg-6.1.2-dev/include/libavcodec/packet.h:543:3: note: declared here
  543 | } AVPacketList;
      |   ^~~~~~~~~~~~
/build/source/extern/osg-ffmpeg-videoplayer/videostate.cpp: In member function 'int Video::PacketQueue::get(AVPacket*, Video::VideoS>
/build/source/extern/osg-ffmpeg-videoplayer/videostate.cpp:144:23: warning: 'AVPacketList' is deprecated [-Wdeprecated-declarations]
  144 |         AVPacketList *pkt1 = this->first_pkt;
      |                       ^~~~
/nix/store/ai243axvzm3y9yq7ap2fl8yidn2a3dj7-ffmpeg-6.1.2-dev/include/libavcodec/packet.h:543:3: note: declared here
  543 | } AVPacketList;
      |   ^~~~~~~~~~~~
/build/source/extern/osg-ffmpeg-videoplayer/videostate.cpp: In member function 'void Video::PacketQueue::clear()':
/build/source/extern/osg-ffmpeg-videoplayer/videostate.cpp:176:19: warning: 'AVPacketList' is deprecated [-Wdeprecated-declarations]
  176 |     AVPacketList *pkt, *pkt1;
      |                   ^~~
/nix/store/ai243axvzm3y9yq7ap2fl8yidn2a3dj7-ffmpeg-6.1.2-dev/include/libavcodec/packet.h:543:3: note: declared here
  543 | } AVPacketList;
      |   ^~~~~~~~~~~~
/build/source/extern/osg-ffmpeg-videoplayer/videostate.cpp:176:25: warning: 'AVPacketList' is deprecated [-Wdeprecated-declarations]
  176 |     AVPacketList *pkt, *pkt1;
      |                         ^~~~
/nix/store/ai243axvzm3y9yq7ap2fl8yidn2a3dj7-ffmpeg-6.1.2-dev/include/libavcodec/packet.h:543:3: note: declared here
  543 | } AVPacketList;
      |   ^~~~~~~~~~~~
/build/source/extern/osg-ffmpeg-videoplayer/videostate.cpp: In member function 'void Video::VideoThread::run()':
/build/source/extern/osg-ffmpeg-videoplayer/videostate.cpp:443:23: warning: 'void av_init_packet(AVPacket*)' is deprecated [-Wdeprec>
  443 |         av_init_packet(&packetData);
      |         ~~~~~~~~~~~~~~^~~~~~~~~~~~~
/nix/store/ai243axvzm3y9yq7ap2fl8yidn2a3dj7-ffmpeg-6.1.2-dev/include/libavcodec/packet.h:629:6: note: declared here
  629 | void av_init_packet(AVPacket *pkt);
      |      ^~~~~~~~~~~~~~
[  4%] Building CXX object components/CMakeFiles/components.dir/resource/resourcesystem.cpp.o
make[2]: *** [components/CMakeFiles/components.dir/build.make:225: components/CMakeFiles/components.dir/vfs/filesystemarchive.cpp.o]>
make[2]: *** Waiting for unfinished jobs....
[  4%] Building CXX object extern/oics/CMakeFiles/oics.dir/ICSInputControlSystem.cpp.o
/build/source/extern/osg-ffmpeg-videoplayer/videostate.cpp: In member function 'void Video::ParseThread::run()':
/build/source/extern/osg-ffmpeg-videoplayer/videostate.cpp:511:23: warning: 'void av_init_packet(AVPacket*)' is deprecated [-Wdeprec>
  511 |         av_init_packet(&packetData);
      |         ~~~~~~~~~~~~~~^~~~~~~~~~~~~
/nix/store/ai243axvzm3y9yq7ap2fl8yidn2a3dj7-ffmpeg-6.1.2-dev/include/libavcodec/packet.h:629:6: note: declared here
  629 | void av_init_packet(AVPacket *pkt);
      |      ^~~~~~~~~~~~~~
[  4%] Building CXX object extern/oics/CMakeFiles/oics.dir/ICSInputControlSystem_keyboard.cpp.o
[  5%] Building CXX object extern/oics/CMakeFiles/oics.dir/ICSInputControlSystem_mouse.cpp.o
[  5%] Building CXX object extern/oics/CMakeFiles/oics.dir/ICSInputControlSystem_joystick.cpp.o
[  5%] Linking CXX static library libosg-ffmpeg-videoplayer.a
[  5%] Built target osg-ffmpeg-videoplayer
[  5%] Linking CXX static library liboics.a
[  5%] Built target oics
make[1]: *** [CMakeFiles/Makefile2:615: components/CMakeFiles/components.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.13.9-xanmod1, NixOS, 25.05 (Warbler), 25.05.20250421.c11863f`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.93.0-dev-pre20250422-7dbdd5b
System type: x86_64-linux
Additional system types: i686-linux, x86_64-v1-linux, x86_64-v2-linux, x86_64-v3-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/ash/.config/nix/nix.conf:/etc/xdg/nix/nix.conf:/home/ash/.local/share/flatpak/exports/etc/xdg/nix/nix.conf:/var/lib/flatpak/exports/etc/xdg/nix/nix.conf:/home/ash/.nix-profile/etc/xdg/nix/nix.conf:/nix/profile/etc/xdg/nix/nix.conf:/home/ash/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/ash/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/5f4w7vp45a46kway0xl0mi3cs5dqxbkz-lix-2.93.0-dev-pre20250422-7dbdd5b/share`
 - nixpkgs: `/nix/store/kcmmd6alr3lx56vkf72503h3pxgf6iv4-source`

### Notify maintainers


@peterhoeg
---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/games/openmw/tes3mp-gcc14-fix.patch
+++ b/pkgs/games/openmw/tes3mp-gcc14-fix.patch
@@ -1,0 +1,25 @@
+diff --git a/apps/openmw/mwstate/charactermanager.hpp b/apps/openmw/mwstate/charactermanager.hpp
+index 2daf734..b77d2a8 100644
+--- a/apps/openmw/mwstate/charactermanager.hpp
++++ b/apps/openmw/mwstate/charactermanager.hpp
+@@ -1,6 +1,8 @@
+ #ifndef GAME_STATE_CHARACTERMANAGER_H
+ #define GAME_STATE_CHARACTERMANAGER_H
+ 
++#include <list>
++
+ #include <boost/filesystem/path.hpp>
+ 
+ #include "character.hpp"
+diff --git a/components/vfs/filesystemarchive.cpp b/components/vfs/filesystemarchive.cpp
+index 6eef4b9..608323e 100644
+--- a/components/vfs/filesystemarchive.cpp
++++ b/components/vfs/filesystemarchive.cpp
+@@ -1,5 +1,7 @@
+ #include "filesystemarchive.hpp"
+ 
++#include <algorithm>
++
+ #include <boost/filesystem.hpp>
+ 
+ #include <components/debug/debuglog.hpp>

--- a/pkgs/games/openmw/tes3mp.nix
+++ b/pkgs/games/openmw/tes3mp.nix
@@ -114,6 +114,9 @@ let
 
       # https://github.com/TES3MP/openmw-tes3mp/issues/552
       ./tes3mp.patch
+
+      # https://github.com/TES3MP/TES3MP/pull/691
+      ./tes3mp-gcc14-fix.patch
     ];
 
     env.NIX_CFLAGS_COMPILE = "-fpermissive";

